### DISCUSSION TITLE
Morphia POJO mapping - used when accessing mongodb driver directly?

### DISCUSSION THREAD

**Discussion by CycleMost** [MorphiaOrg/morphia#2137] [Q&A] [answered]
Based on my testing, it looks like the Morphia pojo mapping (via Entity, Property annotations etc.) is used even when bypassing the morphia classes and using the native mongodb driver. For example:

```
    valuesToSet.add(Updates.set("purchaseDate", new Date()));
    valuesToSet.add(Updates.set("similarCars", similarCars));  // similarCars is a collection of pojo class with Morphia annotations.
    
    Bson updates = Updates.combine(valuesToSet);    
    collection.updateOne(updateQuery, updates);
```
The values updated will used the mapped properties of the pojo object as defined by Morhpia. 

- Correct?
- If so, that is awesome.

Thanks

**Comment by evanchooly** [comment, marked as answer]
Morphia doesn't directly affect the coded registry being used by the driver.  There might be a side effect hidden somewhere in the driver code but as those types are all (largely) immutable I'd be surprised.  If you're seeing such behavior I wouldn't depend on it as that's not an intentional (or actual, afaict) feature and as such might change in the future.
